src/offset/local.rs

```diff
@@ -75,6 +77,263 @@ fn datetime_to_timespec(d: &NaiveDateTime, local: bool) -> sys::Timespec {
+            assert_eq!(naive_date_from_mwd_parts(2022, 1, 2, 3), NaiveDate::from_ymd(2022, 1, 12));
+            assert_eq!(naive_date_from_mwd_parts(2022, 1, 2, 6), NaiveDate::from_ymd(2022, 1, 8));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 0), NaiveDate::from_ymd(2022, 3, 6));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 1), NaiveDate::from_ymd(2022, 3, 7));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 2), NaiveDate::from_ymd(2022, 3, 1));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 3), NaiveDate::from_ymd(2022, 3, 2));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 4), NaiveDate::from_ymd(2022, 3, 3));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 5), NaiveDate::from_ymd(2022, 3, 4));
+            assert_eq!(naive_date_from_mwd_parts(2022, 3, 1, 6), NaiveDate::from_ymd(2022, 3, 5));
+        }
+    }
+
+    fn naive_time_from_seconds(start_time: i32) -> NaiveTime {
+        use std::convert::TryFrom;
+        let h = start_time / 3600;
+        let m = (start_time - 3600 * h) / 60;
+        let s = start_time - 3600 * h - 60 * m;
+        NaiveTime::from_hms(
+            u32::try_from(h).unwrap(),
+            u32::try_from(m).unwrap(),
+            u32::try_from(s).unwrap(),
+        )
+    }
+
+    fn naive_date_from_mwd_parts(year: i32, month: u32, week_num: u32, week_day: u32) -> NaiveDate {
+        // get the first day of the relevant week
+        let base = (week_num.saturating_sub(1)) * 7 + 1;
+
+        // build a date from the first day of the relevant week.
+        // this is the earliest possible date that it will be
+        let base_date = NaiveDate::from_ymd(year, month, base);
+
+        let base_from_sunday = base_date.weekday().num_days_from_sunday();
+
+        use std::cmp::Ordering;
+        match base_from_sunday.cmp(&week_day) {
+            Ordering::Equal => base_date,
+            Ordering::Greater => {
+                base_date + Duration::days(i64::from(7 - (base_from_sunday - week_day)))
+            }
+            Ordering::Less => base_date + Duration::days(i64::from(week_day - base_from_sunday)),
+        }
+    }
+
+    fn naivedatetime_from_mwd(
+        local: NaiveDateTime,
+        mwd: &timezone::MonthWeekDay,
+        start_time: i32,
+    ) -> NaiveDateTime {
+        naive_date_from_mwd_parts(
+            local.year(),
+            local.month(),
+            mwd.week().into(),
+            mwd.week_day().into(),
+        )
+        .and_time(naive_time_from_seconds(start_time))
+    }
+
+    fn utc_now() -> DateTime<Local> {
+        DateTime::<Local>::from_utc(Utc::now().naive_local(), FixedOffset::east(0))
+    }
+
+    fn utc_from_local(local: NaiveDateTime) -> DateTime<Local> {
+        DateTime::<Local>::from_utc(local, FixedOffset::east(0))
+    }
+
+    fn try_now() -> Result<DateTime<Local>, error::TzError> {
+        if path::Path::new(LOCALTIME_LOCATION).exists() {
```

Comment:
Wouldn't we also need a `TZ` environment variable lookup and parse here and other places that use `LOCALTIME_LOCATION`?

Reply:
Hi @MiSawa - thanks for pointing this out. I've added a check for the `TZ` environment variable, and if it exists, use that with preference to the `/etc/localtime` file.

Reply:
Awesome, thank you!